oap-server/server-storage-plugin/storage-banyandb-plugin/src/main/java/org/apache/skywalking/oap/server/storage/plugin/banyandb/BanyanDBConverter.java

```diff
@@ -0,0 +1,265 @@
+            return this.streamWrite;
+        }
+    }
+
+    @Slf4j
+    @RequiredArgsConstructor
+    public static class MeasureToStorage implements Convert2Storage<MeasureWrite> {
+        private final MetadataRegistry.Schema schema;
+        private final MeasureWrite measureWrite;
+
+        @Override
+        public void accept(String fieldName, Object fieldValue) {
+            MetadataRegistry.ColumnSpec columnSpec = this.schema.getSpec(fieldName);
+            if (columnSpec == null) {
+                throw new IllegalArgumentException("fail to find field[" + fieldName + "]");
+            }
+            try {
+                if (columnSpec.getColumnType() == MetadataRegistry.ColumnType.TAG) {
+                    this.measureWrite.tag(fieldName, buildTag(fieldValue, columnSpec.getColumnClass()));
+                } else {
+                    this.measureWrite.field(fieldName, buildField(fieldValue, columnSpec.getColumnClass()));
+                }
+            } catch (BanyanDBException ex) {
+                log.error("fail to add tag", ex);
+            }
+        }
+
+        public void acceptID(String id) {
+            try {
+                this.measureWrite.setID(id);
+            } catch (BanyanDBException ex) {
+                log.error("fail to add ID tag", ex);
+            }
+        }
+
+        @Override
+        public void accept(String fieldName, byte[] fieldValue) {
+            MetadataRegistry.ColumnSpec columnSpec = this.schema.getSpec(fieldName);
+            try {
+                if (columnSpec.getColumnType() == MetadataRegistry.ColumnType.TAG) {
+                    this.measureWrite.tag(fieldName, TagAndValue.binaryTagValue(fieldValue));
+                } else {
+                    this.measureWrite.field(fieldName, TagAndValue.binaryFieldValue(fieldValue));
+                }
+            } catch (BanyanDBException ex) {
+                log.error("fail to add binary tag/field", ex);
+            }
+        }
+
+        @Override
+        public void accept(String fieldName, List<String> fieldValue) {
+            for (final String tagKeyAndValue : fieldValue) {
+                if (StringUtil.isEmpty(tagKeyAndValue)) {
+                    continue;
+                }
+                int pos = tagKeyAndValue.indexOf("=");
+                if (pos == -1) {
+                    continue;
+                }
+                String key = tagKeyAndValue.substring(0, pos);
+                String value = tagKeyAndValue.substring(pos + 1);
+                this.accept(key, value);
+            }
+        }
+
+        @Override
+        public Object get(String fieldName) {
+            return null;
+        }
+
+        @Override
+        public MeasureWrite obtain() {
+            return this.measureWrite;
+        }
+    }
+
+    private static Serializable<BanyandbModel.TagValue> buildTag(Object value, final Class<?> clazz) {
+        if (Integer.class.equals(clazz) || int.class.equals(clazz)) {
```

Comment:
We prefer primary type(int) than object type(Integer), switching these two conditions should be able to speed up.